GVFS/GVFS.Common/Database/PlaceholderTable.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+
+namespace GVFS.Common.Database
+{
+    /// <summary>
+    /// This class is for interacting with the Placeholder table in the SQLite database
+    /// </summary>
+    public class PlaceholderTable : IPlaceholderCollection
+    {
+        private IGVFSConnectionPool connectionPool;
+
+        public PlaceholderTable(IGVFSConnectionPool connectionPool)
+        {
+            this.connectionPool = connectionPool;
+        }
+
+        public static void CreateTable(IDbConnection connection)
+        {
+            using (IDbCommand command = connection.CreateCommand())
+            {
+                command.CommandText = "CREATE TABLE IF NOT EXISTS [Placeholder] (path TEXT PRIMARY KEY, pathType TINYINT NOT NULL, sha char(40) ) WITHOUT ROWID;";
+                command.ExecuteNonQuery();
+            }

```

Comment:
I see now based on the unit tests this is expected ... maybe this is OK? What was the reason for not using `GVFSDatabaseException` here?

Reply:
Didn't want the exception to be double wrapped because this is called from the `Initialize` of `GVFSDatabase` and that is already wrapped in a `try/catch` in the constructor.  The catch and throw of the `GVFSDatabaseException` here would be caught again and wrapped in another `GVFSDatabaseException` in the constructor of `GVFSDatabase`